app/ide-desktop/lib/dashboard/src/modals/TermsOfServiceModal.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * @file
+ *
+ * Modal for accepting the terms of service.
+ */
+
+import * as React from 'react'
+
+import * as reactQuery from '@tanstack/react-query'
+import * as router from 'react-router'
+import invariant from 'tiny-invariant'
+
+import * as authProvider from '#/providers/AuthProvider'
+import * as localStorageProvider from '#/providers/LocalStorageProvider'
+import * as textProvider from '#/providers/TextProvider'
+
+import * as aria from '#/components/aria'
+import * as ariaComponents from '#/components/AriaComponents'
+import Checkbox from '#/components/styled/Checkbox'
+
+import type * as localStorage from '#/utilities/LocalStorage'
+import LocalStorage from '#/utilities/LocalStorage'
+
+declare module '#/utilities/LocalStorage' {
+  /**
+   * The data stored in the `termsOfService` key.
+   */
+  interface AcceptedTosData {
+    /**
+     * The hash of the latest terms of service version that the user has accepted.
+     */
+    readonly versionHash: string
+  }
+
+  /**
+   * Contains the latest terms of service version hash that the user has accepted.
+   */
+  interface LocalStorageData {
+    readonly termsOfService: AcceptedTosData | null
+  }
+}
+
+LocalStorage.registerKey('termsOfService', {
+  tryParse: value => {
+    if (
+      typeof value === 'object' &&
+      value != null &&
+      'versionHash' in value &&
+      typeof value.versionHash === 'string'
+    ) {
+      // eslint-disable-next-line no-restricted-syntax
+      return value as localStorage.LocalStorageData['termsOfService']
+    } else {
+      return null
+    }
+  },
+})
+
+export const latestTermsOfService = reactQuery.queryOptions({
+  queryKey: ['termsOfService', 'currentVersion'],
+  queryFn: () =>
+    fetch(new URL('/eula.json', process.env.ENSO_CLOUD_ENSO_HOST))
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Failed to fetch terms of service')
+        } else {
+          return response.json()
+        }
+      })
+      .then(data => {
+        invariant(data != null, 'Invalid terms of service response')
```

Comment:
@somebody1234 I think it's time to add a schema-validation library, WDYT?

Reply:
we already have ajv as a dependency, is that not sufficient?

Reply:
I guess ajv should be sufficient

Reply:
Uh oh... 
![CleanShot 2024-05-17 at 14 36 36@2x](https://github.com/enso-org/enso/assets/61194245/ffca8636-b61f-41c7-b27e-1cbde18cedb7)

Reply:
Do we need ajv though?